apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -264,7 +340,13 @@ export const pageRouter = router({
 
   getRootPage: protectedProcedure
     .input(getRootPageSchema)
-    .query(async ({ input: { siteId } }) => {
+    .query(async ({ ctx, input: { siteId } }) => {
+      await validateUserPermissions({
```

Comment:
can this authorization be a middleware? if we have a map of the procedures to the required perms?

Reply:
there were a few issues i faced with this (the middleware approach) - we need to tell the middleware _ahead of time_ what permissions are required (`read/create` etc) and also, the middleware requires a few things on the `input` itself (`siteId, resourceId, userId`) which is abit hard because the middleware will impact the downstream typing 

i think what we needed here is a [standalone middleware](https://trpc.io/docs/v10/server/middlewares#experimental-standalone-middlewares) but i didn't do it here cos it's marked as experimental. 

karrui pointed out that we can just use it first (and `starter-kit` does use `unstable_pipe`) but i'm of the opinion that we should get it working w tests first before shifting! wdyt?